main.c

```diff
@@ -114,29 +116,49 @@ int main(int argc, char **argv) {
                 b = 2;
                 x = y = 5;
                 break;
+            /* SELECT PREVIOUS BLOCK */
+            case 44 : /* , */
+            case 60 : /* < */
+                if (b < 3) { /* IF SOME BLOCK IS SELECTED */
+                    oldb = b;
+                    while (!bka[++b%3]) {}
+                    b %= 3;
+                    if (oldb != b) x = y = 5;
+                }
+                break;
+            /* SELECT NEXT BLOCK */
+            case 46 : /* . */
+            case 62 : /* > */
+                if (b < 3) { /* IF SOME BLOCK IS SELECTED */
+                    oldb = b;
+                    while (!bka[++b%3]) {}
+                    b %= 3;
+                    if (oldb != b) x = y = 5;
+                }
+                break;
             case 4 : /* LEFT */
-                if (x > 0
+                if (b < 3 && x > 0
```

Comment:
If this got too repetitive, you could maybe leave these cases in another `switch` executed only if a block is selected.